src/System.CommandLine.Tests/ArgumentTests.cs

```diff
@@ -500,6 +500,42 @@ public void Parse_delegate_is_called_once_per_parse_operation()
                 i.Should().Be(2);
             }
 
+            [Theory] // https://github.com/dotnet/command-line-api/issues/1294
+            [InlineData("", "option-is-implicit")]
+            [InlineData("--bananas", "argument-is-implicit")]
+            [InlineData("--bananas argument-is-specified", "argument-is-specified")]
+            public void Parse_delegate_is_called_when_Option_Arity_allows_zero_tokens(string commandLine, string expectedValue)
+            {
+                var opt = new Option<string>(
+                    "--bananas",
+                    parseArgument: result =>
```

Comment:
@jonsequitur what should be returned when the option cannot be parsed? 
Example: argument supposed to be implicit, but we don't have value for this defined. I would like to fallback to default behavior (parse error) in this case.

Reply:
Set the `ArgumentResult.ErrorMessage` property to the error you want to display.